src/common/transformations/src/transformations/common_optimizations/eliminate_loop_inputs_outputs.cpp

```diff
@@ -0,0 +1,185 @@
+    MATCHER_SCOPE(EliminateLoopInputsOutputs);
+
+    auto subgraph_label = wrap_type<ov::op::v5::Loop, ov::op::v0::TensorIterator>();
+
+    matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        using namespace ov::op::util;
+        using InvariantD = MultiSubGraphOp::InvariantInputDescription;
+        using SlicedD = MultiSubGraphOp::SliceInputDescription;
+        using MergedD = MultiSubGraphOp::MergedInputDescription;
+        using OutputD = MultiSubGraphOp::BodyOutputDescription;
+        using ConcatD = MultiSubGraphOp::ConcatOutputDescription;
+
+        // deleting parameter/results are inserted to the beginning of deque, adding Descriptions to the end
+        // deque: delete_param, delete_param, add_input_desc, add_inputs_desc ...
+        // delete_param updates Parameter index in the vector, add_input_desc will use the updated indexes.
+        std::deque<std::function<void()>> delete_inputs;
+        std::deque<std::function<void()>> delete_outputs;
+        std::deque<std::function<void()>> process_inputs;
+        std::deque<std::function<void()>> process_outputs;
+
+        const auto& pattern_to_output = m.get_pattern_value_map();
+
+        auto subgraph =
+            as_type_ptr<ov::op::util::SubGraphOp>(pattern_to_output.at(subgraph_label).get_node_shared_ptr());
+        if (!subgraph) {
+            return false;
+        }
+        const auto& body_model = subgraph->get_function();
+        const auto& body_params = body_model->get_parameters();
+        const auto& body_results = body_model->get_results();
+
+        std::shared_ptr<SubGraphOp> new_node;
+        const auto& subgraph_in_values = subgraph->input_values();
+        int64_t body_condition_output_idx = -1;
+        if (auto loop = as_type_ptr<ov::op::v5::Loop>(subgraph)) {
+            const auto& trip_count = subgraph_in_values[0];
+            const auto& exec_cond = subgraph_in_values[1];
+
+            auto new_loop = make_shared<ov::op::v5::Loop>(trip_count, exec_cond);
+            new_loop->set_special_body_ports(loop->get_special_body_ports());
+            new_node = new_loop;
+            body_condition_output_idx = loop->get_special_body_ports().body_condition_output_idx;
+        } else {
+            new_node = make_shared<ov::op::v0::TensorIterator>();
+        }
+        new_node->set_function(body_model);
+
+        for (const auto& input_description : subgraph->get_input_descriptions()) {
+            const auto& body_param = body_params[input_description->m_body_parameter_index];
+            const auto& init_value = subgraph_in_values[input_description->m_input_index];
+
+            if (const auto merged_input_desc = as_type_ptr<MergedD>(input_description)) {
+                const auto& body_res = body_results[merged_input_desc->m_body_value_index];
+                auto param = get_parent_param(body_res);
+                if (param && param->get_output_target_inputs(0).size() == 1) {
```

Comment:
If there will be body subgraph with Parameter that connected to several (>1) Results and has no other consumers (not Result consumers), it will no be deleted? Is it expected?

Reply:
I should say that it is not a case for customer model.
Also `lstm = tf.keras.layers.LSTM(1, return_sequences = True, return_state = True)` cannnot be fused to LSTMSequence since it just returns cell state for the last step (not a sequence).

Reply:
>> Parameter that connected to several (>1) Results and has no other consumers (not Result consumers), it will no be deleted? Is it expected
I hope we don't really have such subgraphs
Parameter - > Result_1
                 - >  Result_2
No reason to have the 2nd Result in this case.

Reply:
Fixed. Added handling a case of Parameter going to multiple Result nodes